test/functional/plugin/lsp/diagnostic_spec.lua

```diff
@@ -497,6 +498,85 @@ describe('vim.lsp.diagnostic', function()
           return _G.params.previousResultId
         end)
       )
+
+      -- Unchanged reports
+      eq(
+        'squidward',
+        exec_lua(function()
+          vim.lsp.diagnostic.on_diagnostic(nil, {
+            kind = 'unchanged',
+            resultId = 'squidward',
+          }, {
+            method = vim.lsp.protocol.Methods.textDocument_diagnostic,
+            params = {
+              textDocument = { uri = fake_uri },
+            },
+            client_id = client_id,
+            bufnr = diagnostic_bufnr,
+          })
+          vim.api.nvim_exec_autocmds('LspNotify', {
+            buffer = diagnostic_bufnr,
+            data = {
+              method = vim.lsp.protocol.Methods.textDocument_didChange,
+              client_id = client_id,
+            },
+          })
+          return _G.params.previousResultId
+        end)
+      )
+    end)
+
+    it('handles related document diagnostics', function()
```

Comment:
helpful for searching if we use the literal protocol name
```suggestion
    it('handles relatedDocuments diagnostics', function()
```